spec/jobs/whitehall_document_import_job_spec.rb

```diff
@@ -46,61 +49,90 @@
+    let(:error_message) { "import failed error" }
     before do
-      allow(WhitehallImporter::Import).to receive(:call).and_raise(error)
+      allow(WhitehallImporter::Import).to receive(:call).and_raise(error_message)
     end
 
-    it "retries the job" do
-      WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
+    it "unlocks the document in Whitehall" do
+      request = stub_whitehall_api_unlock_document(
+        whitehall_migration_document_import.whitehall_document_id,
+      )
 
-      expect(WhitehallDocumentImportJob).to have_been_enqueued
+      WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
+      expect(request).to have_been_requested
     end
 
-    it "logs the error when retries have been exhausted" do
-      perform_enqueued_jobs do
-        WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
-      end
+    it "updates the document import state to 'import_failed'" do
+      WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
 
       whitehall_migration_document_import.reload
-      expect(whitehall_migration_document_import.error_log).to eq(error.inspect)
+      expect(whitehall_migration_document_import).to be_import_failed
     end
+  end
 
-    it "updates the document import state to 'import_failed' if the import failed" do
-      perform_enqueued_jobs do
-        WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
-      end
+  context "when the sync fails" do
+    let(:error_message) { "sync failed error" }
+    before do
+      allow(WhitehallImporter::Import).to receive(:call)
+      allow(WhitehallImporter::Sync).to receive(:call).and_raise(error_message)
+    end
 
-      whitehall_migration_document_import.reload
-      expect(whitehall_migration_document_import).to be_import_failed
+    it "does not unlock the document in Whitehall" do
+      request = stub_whitehall_api_unlock_document(
+        imported_document_import.whitehall_document_id,
+      )
+
+      WhitehallDocumentImportJob.perform_now(imported_document_import)
+      expect(request).not_to have_been_requested
+    end
+
+    it "updates the document import state to 'sync_failed'" do
+      WhitehallDocumentImportJob.perform_now(imported_document_import)
+
+      imported_document_import.reload
+      expect(imported_document_import).to be_sync_failed
     end
+  end
 
-    it "updates the document import state to 'sync_failed' if the sync failed" do
-      whitehall_migration_document_import.update!(state: "imported")
+  context "when a GdsApi::BaseError exception is raised" do
+    let(:error) { GdsApi::BaseError.new }
+    before do
+      allow(WhitehallImporter::Import).to receive(:call).and_raise(error)
+    end
 
+    it "retries the job" do
+      WhitehallDocumentImportJob.perform_now(whitehall_migration_document_import)
+
+      expect(WhitehallDocumentImportJob).to have_been_enqueued
```

Comment:
Does this actually test the job has been retried rather than just queued originally?

Reply:
I believe so.  If you look at the errors on which we don't retry, we test that the job has not been enqueued e.g. https://github.com/alphagov/content-publisher/blob/master/spec/jobs/whitehall_document_import_job_spec.rb#L148

Reply:
Ah ha cool, just checking! 👍